pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,320 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchzip,
+  premake5,
+  writeShellScriptBin,
+  runCommandLocal,
+  symlinkJoin,
+  imagemagick,
+  bzip2,
+  curl,
+  flac,
+  # Use fmt 10+ after release 40.1.4+
+  fmt_9,
+  freetype,
+  libevent,
+  libgit2,
+  libGL,
+  libGLU,
+  libjpeg,
+  libpng,
+  libvorbis,
+  libX11,
+  libxkbcommon,
+  libXxf86vm,
+  lua5_3,
+  mono,
+  nlohmann_json,
+  openal,
+  SDL2,
+  sqlite,
+  wayland,
+  egl-wayland,
+  covers_url ? "https://pics.projectignis.org:2096/pics/cover/{}.jpg",
+  fields_url ? "https://pics.projectignis.org:2096/pics/field/{}.png",
+  # While ygoprodeck has higher quality images, "spamming" of their api results in a ban.
+  # Thats why this link can change since it's compiled into the program, However it will
+  # download assets when needed so it is unlikely to get banned. Unless you search the
+  # card list with no filters of any kind. When testing use ProjectIgnis' website instead.
+  pics_url ? "https://images.ygoprodeck.com/images/cards/{}.jpg",
+}:
+let
+  archLabel =
+    {
+      "x86_64-linux" = "x64";
+      "aarch64-linux" = "arm64";
+    }
+    .${stdenv.hostPlatform.system}
+      or (throw "Don't know how to translate ${stdenv.hostPlatform.system} into an arch label for edopro");
+
+  deps = import ./deps.nix;
+in
+let
+  assets = {
+    pname = "assets";
+    version = deps.edopro-version;
+
+    src = fetchzip {
+      url = "https://github.com/ProjectIgnis/edopro-assets/releases/download/${deps.edopro-version}/ProjectIgnis-EDOPro-${deps.edopro-version}-linux.tar.gz";
+      sha256 = deps.assets-hash;
+    };
+  };
+
+  irrlicht-edopro = stdenv.mkDerivation {
+    pname = "irrlicht-edopro";
+    version = "1.9-custom";
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = "irrlicht1-8-4";
+      rev = deps.irrlicht-rev;
+      hash = deps.irrlicht-hash;
+    };
+
+    buildInputs = [
+      libGLU
+      libX11
+      libxkbcommon
+      libXxf86vm
+      wayland
+    ];
+
+    enableParallelBuilding = true;
+    buildFlags = [
+      "NDEBUG=1"
+    ];
+    makeFlags = [
+      "-C"
+      "source/Irrlicht"
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/{bin,include}
+      cp lib/Linux/libIrrlicht.a $out/bin
+      cp -r include/* $out/include
+
+      runHook postInstall
+    '';
+  };
+
+  ocgcore = stdenv.mkDerivation rec {
+    pname = "ygopro-core";
+    version = deps.ocgcore-rev;
```

Comment:
Similar note here, except that this follows our version guideline even less than `irrlicht-edopro` - `"4a872ad7686e0d00ee4c1c051e90fc07c40613d4"`.

Currently this points at https://github.com/edo9300/ygopro-core/commit/4a872ad7686e0d00ee4c1c051e90fc07c40613d4, which would be `9.1-unstable-2022-09-15`. [There was a release 2 days after that commit](https://github.com/edo9300/ygopro-core/releases/tag/v10.0) which would make it a clean `10.0`, but dunno if they didn't bump to that on purpose (maybe worth a ping upstream?).

Reply:
This is a git module so the context of versions doesn't really make sense here. Even if I update the script to fetch the commit and see if it has a related tag its not always going to resolve correctly.

Reply:
I pinged him anyways

Reply:
How on earth do you want my script to get the version this core is from?